internal/services/machinelearning/machine_learning_workspace_network_outbound_rule_service_tag_resource.go

```diff
@@ -0,0 +1,344 @@
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model machineLearningWorkspaceServiceTagOutboundRuleModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.MachineLearning.ManagedNetwork
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			workspaceId, err := managednetwork.ParseWorkspaceID(model.WorkspaceId)
+			if err != nil {
+				return err
+			}
+			id := managednetwork.NewOutboundRuleID(subscriptionId, workspaceId.ResourceGroupName, workspaceId.WorkspaceName, model.Name)
+			existing, err := client.SettingsRuleGet(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return tf.ImportAsExistsError("azurerm_machine_learning_workspace_network_outbound_rule_service_tag", id.ID())
+			}
+
+			outboundRule := managednetwork.OutboundRuleBasicResource{
+				Name: pointer.To(model.Name),
+				Type: pointer.To(string(managednetwork.RuleTypeServiceTag)),
+				Properties: managednetwork.ServiceTagOutboundRule{
+					Category: pointer.To(managednetwork.RuleCategoryUserDefined),
+					Destination: &managednetwork.ServiceTagDestination{
+						PortRanges: &model.PortRanges,
+						Protocol:   &model.Protocol,
+						ServiceTag: &model.ServiceTag,
+					},
+				},
+			}
+
+			if err = client.SettingsRuleCreateOrUpdateThenPoll(ctx, id, outboundRule); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r WorkspaceNetworkOutboundRuleServiceTag) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model machineLearningWorkspaceServiceTagOutboundRuleModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.MachineLearning.ManagedNetwork
+			id, err := managednetwork.ParseOutboundRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.SettingsRuleGet(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", id)
+			}
+
+			if existing.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `properties` was nil", id)
+			}
+
+			payload := existing.Model
+
+			serviceTagOutboundRule := managednetwork.ServiceTagOutboundRule{
+				Type:        managednetwork.RuleTypeServiceTag,
+				Category:    pointer.To(managednetwork.RuleCategoryUserDefined),
+				Destination: &managednetwork.ServiceTagDestination{},
+			}
```

Comment:
Why do we need to define this, it should already exist under `payload.Properties`